statements: store imported statements in redux

sjbarag opened the pull request:
Keeping imported statements in a single location means several components can add, modify, or delete those without duplicating logic or passing functions down as props.  As a convenient side-effect, importing a redux state via Redux DevTools makes for easy debugging :)

The current state is a bit messy, in that imported statements are _also_ stored in the `ImportPage`'s local state.  This is a necessary duplication of data for now, as statement `<textarea>`s result in updates to the component-local state.  A future PR will make that flow through Redux as well, but this is just a PR to unblock other devs 😃

sjbarag committed:
statements: read statements from redux store

sjbarag committed:
statements: control statement (soft) deletion via redux

sjbarag committed:
statements: enable statement restoration after deletion

sjbarag committed:
statements: handle statement insertion via redux

sjbarag committed:
statements: don't pass setShowSQLExec callback into statement

Now that the Statement component is redux-aware, it can dispatch the
"showRawSql" action by itself without requiring an external component to
provide that logic. Stop providing a dialog-show callback to individual
statements.

sjbarag committed:
layout: move import components to features/imports/

Now that the import-related components are connected to the Redux store,
they're considered part of a single "feature" and should no longer be in
app/.  Move the import page and its related components into
features/imports/.

otan commented on src/features/imports/importsSlice.ts:
```diff
@@ -48,6 +49,24 @@ export const importsSlice = createSlice({
+      }
+
+      const modifiedStatements = payload.map(stmt => ({
+        ...stmt,
```
can we also remove the `unimplemented` issues associated with it when we do this?
this resets the summary appropriately.
i don't mind it not reappearing when it gets undone.

sjbarag replied:
Oooh I forgot to have the summary consider statement deletion!  That should be pretty easy to do, one moment.

sjbarag replied:
Done!

nathanstilwell commented on src/features/imports/importsSlice.ts:
```diff
@@ -0,0 +1,107 @@
+import { createSlice, createEntityAdapter, PayloadAction, EntityState, nanoid } from "@reduxjs/toolkit";
```
Nice. I didn't realize that `@reduxjs/toolkit` comes with `nanoid`.

nathanstilwell reviewed (approved):
LGTM. Nice work with redux toolkit.

sjbarag committed:
lint: remove spurious useCallback in dialogs

Because `dispatch` is stable, there's no need to wrap a callback that
only calls `dispatch` in `useCallback`.
